lotus-soup/testkit/role_miner_sync.go

```diff
@@ -0,0 +1,169 @@
+package testkit
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/specs-actors/actors/abi"
+	"github.com/testground/sdk-go/sync"
+)
+
+// MinerSyncTopic is the topic where miner events are exchanged.
+var MinerSyncTopic = sync.NewTopic("miner_sync", &MiningSyncMsg{})
+
+// MiningSyncMsg represents a miner event.
```

Comment:
```suggestion
// MiningSyncMsg signals that the miner attempted to mine in a particular epoch (whether successful or not).
```

Feel free to change the wording, what I'd like to achieve is more specificity in the description, this event is triggered after running `mineRound`/`MineOne`.

Reply:
This is much better, although `successful` implies that a false value here would be a _failure`, which it isn't. We attempted to _mine_ successfully, but we weren't elected so we didn't produce a block. Will fine-tune.